forge-ai/src/main/java/forge/ai/ComputerUtil.java

```diff
@@ -2544,35 +2544,36 @@ public static String chooseSomeType(Player ai, String kindOfType, SpellAbility s
 
                 if (logic.equals("MostProminentOnBattlefield")) {
                     chosen = ComputerUtilCard.getMostProminentType(game.getCardsIn(ZoneType.Battlefield), valid);
-                }
-                else if (logic.equals("MostProminentComputerControls")) {
+                } else if (logic.equals("MostProminentComputerControls")) {
                     chosen = ComputerUtilCard.getMostProminentType(ai.getCardsIn(ZoneType.Battlefield), valid);
-                }
-                else if (logic.equals("MostProminentComputerControlsOrOwns")) {
+                } else if (logic.equals("MostProminentComputerControlsOrOwns")) {
                     CardCollectionView list = ai.getCardsIn(Arrays.asList(ZoneType.Battlefield, ZoneType.Hand));
                     if (list.isEmpty()) {
                         list = ai.getCardsIn(Arrays.asList(ZoneType.Library));
                     }
                     chosen = ComputerUtilCard.getMostProminentType(list, valid);
-                }
-                else if (logic.equals("MostProminentOppControls")) {
+                } else if (logic.equals("MostProminentOppControls")) {
                     CardCollection list = ai.getOpponents().getCardsIn(ZoneType.Battlefield);
                     chosen = ComputerUtilCard.getMostProminentType(list, valid);
                     if (!CardType.isACreatureType(chosen) || invalidTypes.contains(chosen)) {
                         list = CardLists.filterControlledBy(game.getCardsInGame(), ai.getOpponents());
                         chosen = ComputerUtilCard.getMostProminentType(list, valid);
                     }
-                }
-                else if (logic.startsWith("MostProminentInComputerDeck")) {
+                } else if (logic.startsWith("MostProminentInComputerDeck")) {
                     boolean includeTokens = !logic.endsWith("NonToken");
                     chosen = ComputerUtilCard.getMostProminentType(ai.getAllCards(), valid, includeTokens);
-                }
-                else if (logic.equals("MostProminentInComputerGraveyard")) {
+                } else if (logic.equals("MostProminentInComputerGraveyard")) {
                     chosen = ComputerUtilCard.getMostProminentType(ai.getCardsIn(ZoneType.Graveyard), valid);
                 }
             }
+
+            if (sa.hasParam("Note") && sa.getHostCard().hasAnyNotedType())
```

Comment:
this part shouldn't be needed:
the normal assumption is the api effect of the engine will already provide the legal options with its logic, so AI code can focus on logic around good choice etc.

Reply:
I've checked it. The types are the remaining. eg: human wizard and human already noted leaves wizard left.
It'll be removed.

Reply:
removed